File: Next greater element using stack.java
import java.util.Scanner;

class Solution {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter the size of the array");
        int n = sc.nextInt();
        int[] arr = new int[n];
        System.out.println("Enter the elements");

        for (int i = 0; i < n; i++)
            arr[i] = sc.nextInt();

        myStack stack = new myStack(n);
        int[] result = new int[n];

        for (int i = n - 1; i >= 0; i--) {
            while (!stack.isEmpty() && stack.peek() <= arr[i]) {
                stack.pop();
            }

            result[i] = stack.isEmpty() ? -1 : stack.peek();
            stack.push(arr[i]);
        }

        System.out.println("Next Greater Elements:");
        for (int val : result)
            System.out.print(val + " ");
    }
}

File: Next greater element using Stack.py
class Solution:
    def nextLargerElement(self, arr):
        stack = []
        nge = []

        for i in range(len(arr) - 1, -1, -1):
            
            while stack and arr[i] >= stack[-1] :
                stack.pop()
            
            if stack:
                nge.append(stack[-1])
            else:
                nge.append(-1)

            stack.append(arr[i])
        
        return nge[::-1] 


if __name__ == "__main__":
    arr = [3, 10, 4, 2, 1, 2, 6, 1, 7, 2, 9]
    sol = Solution()
    result = sol.nextLargerElement(arr)
    print("Next Greater Elements:", result)